File: src/main/kotlin/com/cout970/magneticraft/api/core/INode.java
package com.cout970.magneticraft.api.core;

import net.minecraft.nbt.CompoundNBT;
import net.minecraftforge.common.util.INBTSerializable;

/**
 * Created by cout970 on 16/06/2016.
 */
public interface INode extends INBTSerializable<CompoundNBT>, ITileRef {

    /**
     * Unique identifier of this node
     */
    NodeID getId();
}

File: src/main/kotlin/com/cout970/magneticraft/api/energy/INode.kt
package com.cout970.magneticraft.api.energy

import net.minecraft.nbt.NBTTagCompound
import net.minecraft.util.math.BlockPos
import net.minecraft.world.World
import net.minecraftforge.common.util.INBTSerializable

/**
 * Created by cout970 on 11/06/2016.
 */
interface INode : INBTSerializable<NBTTagCompound>{

    fun getWorld(): World

    fun getPos(): BlockPos
}